airbyte-integrations/connectors/source-smartsheets/unit_tests/test_source.py

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import json
+from itertools import permutations
+from unittest.mock import Mock
+
+import pytest
+from source_smartsheets.source import SourceSmartsheets
+
+
+@pytest.fixture
+def config():
+    return {"access_token": "token", "spreadsheet_id": "id"}
+
+
+@pytest.fixture(name="catalog")
+def configured_catalog():
+    stream_mock = Mock()
+    stream_mock.name = "test"  # cannot be used in __init__
+    return Mock(streams=[Mock(stream=stream_mock)])
+
+
+columns = [
+    {"id": "1101932201830276", "title": "id", "type": "TEXT_NUMBER"},
+    {"id": "5605531829200772", "title": "first_name", "type": "TEXT_NUMBER"},
+    {"id": "3353732015515524", "title": "last_name", "type": "TEXT_NUMBER"},
+]
+
+
+cells = [
+    {"columnId": "1101932201830276", "value": "11"},
+    {"columnId": "5605531829200772", "value": "Leonardo"},
+    {"columnId": "3353732015515524", "value": "Dicaprio"},
+]
+
+
```

Comment:
add 2 more tests:

Reply:
1. with empty cells
2. with empty cells and columns

Reply:
done